mscore/resourceManager.cpp

```diff
@@ -373,6 +970,232 @@ void ResourceManager::downloadExtension()
             }
       }
 
+static inline QString filenameBaseFromPageURL(QString page_url)
+      {
+      Q_ASSERT(page_url.contains("/project/"));
+      if (page_url.contains("/en/project/"))
+            return page_url.remove("/en/project/");
+      return page_url.remove("/project/");
+      }
+static inline QString getExtFromURL(QString& direct_link)
+      {
+      QString possible_ext = direct_link.split(".").last();
+      // We can assume that most extensions contain only 
+      // alphanumeric and plus "_", and other connector punctuation chars
+      if (possible_ext.contains(QRegularExpression("\\W")))
+            return {};
+      return possible_ext;
+      }
+
+QString ResourceManager::downloadPluginPackage(PluginPackageDescription& desc, const QString& page_url)
+      {
+      DownloadUtils package(QNetworkRequest::RedirectPolicy::NoLessSafeRedirectPolicy, this);
+      package.setTarget(desc.direct_link);
+      // TODO: try to get extension name from direct_link, and add it to localPath
+      QString localPath = dataPath + "/plugins/" + filenameBaseFromPageURL(page_url);
+      if (desc.source == GITHUB || desc.source == GITHUB_RELEASE)
+            localPath += ".zip";
+      else if (desc.source == ATTACHMENT)
+            localPath += "."+QFileInfo(desc.direct_link).suffix();
+      QDir().mkpath(dataPath + "/plugins");
```

Comment:
IIRC "plugins' is translated, so we'd need to pick the translated name for it

Reply:
The "plugins" directory seems to be hard-coded. See https://github.com/musescore/MuseScore/blob/ea81a9338c2e21070b646a14ed9cd8b8d8e0039a/mscore/plugin/mscorePlugins.cpp#L376.

Reply:
The global, system one, yes, but not the user one

Reply:
OK, I've changed the directory path to the user one, which can be acquired via `preferences.getString(PREF_APP_PATHS_MYPLUGINS)`, and has already been translated.